src/Image.js

```diff
@@ -68,16 +68,16 @@ function ImageBase({
   imgProps = {}, 
   layoutClassName = undefined 
 }) {
-  if (process.env.NODE_ENV !== 'production' && !image.asset.metadata) {
-    console.error('Image asset doesn\'t have associated metadata. Did you forget to dereference the asset field (`image{..., asset->}`)?')
+  if (image?.asset && !image.asset.metadata) {
+    throw new Error('Image asset doesn\'t have associated metadata. Did you forget to dereference the asset field (`image{..., asset->}`)?')
   }
 
   const { ref: sizeRef, size: displaySize } = useElementSize()
   const { src, srcSet } = useSrcSet({ config: sanityConfig, image, adjustImage })
   const { width, height, size } = useDerivedSizes({
     deriveSizes, 
     displaySize,
-    naturalSize: image.asset.metadata.dimensions  
+    naturalSize: image.asset?.metadata.dimensions ?? { width: 0, height: 0 }
```

Comment:
Als `asset.metadata` er is... is `asset.metadata.dimensions` er dan ook gegarandeerd?

```suggestion
    naturalSize: image.asset?.metadata?.dimensions ?? { width: 0, height: 0 }
```

Reply:
Right, jullie hebben helemaal gelijk :+1: